### ISSUE TITLE
Google style: Missing support for multipart versioning format using underscores in method names

### ISSUE THREAD

**Issue by mohitsatr** [checkstyle/checkstyle#17507]
From: https://google.github.io/styleguide/javaguide.html#s5.3-camel-case
> In very rare circumstances (for example, multipart version numbers), you may need to use underscores to separate adjacent numbers, since numbers do not have upper and lower case variants. 

Correct version: `guave33_4_6`
incorrect version: `guava3346`

Code:
```java
/** some javadoc. */
public class Test {

  String guava33_4_6 = "Guava 33.4.6"; // False-positive
  String guava3346 = "Guava 33.4.6";   // False-negative 
  
  String jdk8_0_392 = "Jdk 8.0.392";   // False-positive
  String jdk80392 = "Jdk 8.0.392";     // False-negative

  String kotlin1_9_24 = "Kotlin 1.9.24"; // False-positive
  String kotlin1924 = "Kotlin 1.9.24";   // False-negative

  String gradle8_5_1 = "Gradle 8.5.1";   // False-positive
  String gradle851 = "Gradle 8.5.1";     // False-negative
}

```

Cli:
```
$ java -jar checkstyle-10.26.1-all.jar -c google_checks.xml Test.java 
Starting audit...
[WARN] /mnt/5D92528E6B945467/test/testing/Test.java:4:10: Member name 'guava33_4_6' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MemberName]
[WARN] /mnt/5D92528E6B945467/test/testing/Test.java:7:10: Member name 'jdk8_0_392' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MemberName]
[WARN] /mnt/5D92528E6B945467/test/testing/Test.java:10:10: Member name 'kotlin1_9_24' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MemberName]
[WARN] /mnt/5D92528E6B945467/test/testing/Test.java:13:10: Member name 'gradle8_5_1' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MemberName]
Audit done.
```
The style guide doesn't mention this explicitly, and we can have a discussion about it, but I think it's reasonable to assume that such naming conventions are likely intended only for member names.

After adding support for underscores between digits, coverage table should be updated.

**Comment by Zopsss** [comment]
We can allow underscores between digits but regex maybe very tricky for such cases. @mohitsatr

**Comment by Zopsss** [comment]
First let's focus on the false-positives. As they may annoy users. False-negatives are low priority

**Comment by romani** [comment]
It might be good to split issue into two, to let us address false positives first, and false-negatives later.

**Comment by romani** [comment]
>  String guava3346 = "Guava 33.4.6";   // False-negative 

this is ok actaully, we never know what right number is,  it is "guava 3346" , good name

**Comment by romani** [comment]
basic implementation is merged
all false negatves will be addressed in next issue

@mohitsatr , please create more exact issue on what is gap in coverage.
